examples/http/client.cpp

```diff
@@ -0,0 +1,125 @@
+// Simple HTTP client that prints the response
+
+#include "caf/net/http/client.hpp"
+
+#include "caf/net/http/with.hpp"
+#include "caf/net/ip.hpp"
+#include "caf/net/middleman.hpp"
+#include "caf/net/octet_stream/transport.hpp"
+#include "caf/net/tcp_stream_socket.hpp"
+
+#include "caf/actor_system.hpp"
+#include "caf/actor_system_config.hpp"
+#include "caf/caf_main.hpp"
+#include "caf/detail/latch.hpp"
+#include "caf/event_based_actor.hpp"
+#include "caf/ipv4_address.hpp"
+#include "caf/scheduled_actor/flow.hpp"
+#include "caf/span.hpp"
+#include "caf/uuid.hpp"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+#include <utility>
+
+using namespace std::literals;
+
+namespace http = caf::net::http;
+namespace ssl = caf::net::ssl;
+using namespace caf;
+
+// -- constants ----------------------------------------------------------------
+
+// static constexpr uint16_t default_port = 7788;
+// static constexpr std::string_view default_host = "localhost";
+
+constexpr auto default_method = http::method::get;
+
+// -- configuration setup ------------------------------------------------------
+struct config : actor_system_config {
+  config() {
+    opt_group{custom_options_, "global"}
+      .add<http::method>("method,m", "HTTP method to use")
+      .add<uri>("resource,r", "Requested resource (URL)")
+      .add<std::string>("payload,p", "Optional payload to send");
+    opt_group{custom_options_, "tls"} //
+      .add<bool>("enable", "enables encryption via TLS")
+      .add<std::string>("ca-file", "CA file for trusted servers");
+  }
+
+  caf::settings dump_content() const override {
+    auto result = actor_system_config::dump_content();
+    caf::put_missing(result, "method", default_method);
+    caf::put_missing(result, "payload", "");
+    return result;
+  }
+};
+
+auto defatul_port(std::string_view scheme) {
+  if (scheme == "http")
+    return 80;
+  else if (scheme == "https")
+    return 443;
+  else
+    return 0;
+}
+
+int caf_main(caf::actor_system& sys, const config& cfg) {
+  auto maybe_resource = caf::get_as<uri>(cfg, "resource");
+  if (!maybe_resource) {
+    std::cerr << "*** missing mandatory option --resource" << std::endl;
+    return EXIT_FAILURE;
+  }
+  auto resource = std::move(*maybe_resource);
+  auto use_ssl = caf::get_or(cfg, "tls.enable", false);
+  auto ca_file = caf::get_as<std::string>(cfg, "tls.ca-file");
+  if (resource.scheme() != "http" && resource.scheme() != "https") {
+    std::cerr << "*** only HTTP is supported at the moment" << std::endl;
+    return EXIT_FAILURE;
+  }
+  auto method = caf::get_or(cfg, "method", default_method);
+  auto payload = caf::get_or(cfg, "payload", ""sv);
+  auto& auth = resource.authority();
+  auto port = auth.port != 0 ? auth.port : defatul_port(resource.scheme());
+  auto conn
```

Comment:
This does not represent a connection. It's a future for the response plus the disposable for canceling the request. If nothing better comes to mind, `result` would still be a better name.